app/src/main/kotlin/com/wire/android/mapper/RegularMessageContentMapper.kt

```diff
@@ -91,6 +92,34 @@ class RegularMessageMapper @Inject constructor(
             message.deliveryStatus
         )
 
+        is MessageContent.Composite -> {
+            val text = content.textContent?.let { textContent ->
+                val quotedMessage = textContent.quotedMessageDetails?.let { mapQuoteData(message.conversationId, it) }
+                    ?: if (textContent.quotedMessageReference?.quotedMessageId != null) {
+                        UIQuotedMessage.UnavailableData
+                    } else {
+                        null
+                    }
+
+                MessageBody(
+                    message = UIText.DynamicString(textContent.value, content.textContent?.mentions.orEmpty()),
+                    quotedMessage = quotedMessage
+                )
+            }
+
+            UIMessageContent.Composite(
+                messageBody = text,
+                buttonList = content.buttonList.map {
+                    MessageButton(
+                        id = it.id,
+                        text = it.text,
+                        isSelected = it.isSelected
+                    )
+                },
+                deliveryStatus = mapRecipientsFailure(userList, message.deliveryStatus)
```

Comment:
We can assume failure in general for composite messages in federated environments. Or in other words, we are required to show partial delivery failures only for: text messages, assets, and audio messages.

You can take a look at `Knock` for example or likes, no errors are displayed to the user.